src/Engine/Engine.cpp

```diff
@@ -231,6 +231,10 @@ void Engine::DrawGUI() {
         uCurrentlyLoadedLevelType == LEVEL_Outdoor)
         pWeather->Draw();  // Ritor1: my include
 
+    if (current_screen_type != CURRENT_SCREEN::SCREEN_GAME) {
+        pAudioPlayer->stopWalkingSounds();
+    }
```

Comment:
Are you sure this belongs to Engine::DrawGUI?

Reply:
Why not do this in ProcessPartyActions, where all walk sound processing is?

Reply:
Because game menu loop does not calls this function.

Reply:
OK let's slap a TODO here, something like `TODO: this really doesn't belong to Engine::DrawGUI`, and merge.